plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/Inline.hs

```diff
@@ -326,14 +326,19 @@ sizeIsAcceptable = \case
   Delay _ t  -> sizeIsAcceptable t
 
 isPure :: Term name uni fun a -> Bool
-isPure = go
+isPure = go True
     where
-        go = \case
-            Var {}      -> True
+        go delayIsPure = \case
```

Comment:
The `delayIsPure` thing at _least_ needs a note. In fact I think it's unsound. Consider:

```
(\x . force x) (delay impure)
```

Your analysis considers the `force x` to be pure but it isn't.

I'm going to be annoying again and say that you'd have noticed this if you had a) an argument for why it was sound, b) some tests illustrating the new behaviour you were trying to create :)

I'd be more convinced by a case for `force (delay x)`, even though that will definitely get optimized away, since it mirrors the case for `(\x . b) t`.

Reply:
Updated and added some tests.